de-lib/src/main/java/org/iplantc/de/pipelines/client/views/PipelineAppOrderViewImpl.java

```diff
@@ -115,6 +141,21 @@ public PipelineTask getOrderGridSelectedApp() {
         return appOrderGrid.getSelectionModel().getSelectedItem();
     }
 
+    private void setButtonState() {
+        int size = appOrderGrid.getSelectionModel().getSelectedItems().size();
+        if(size > 0) {
+            removeAppBtn.enable();
+            if(appOrderGrid.getStore().size() > 1) {
+                moveDownBtn.enable();
+                moveUpBtn.enable();
+            }
+        } else {
+            removeAppBtn.disable();
+            moveUpBtn.disable();
+            moveDownBtn.disable();
+        }
+    }
```

Comment:
` if(size > 0)` checks if any app is selected. Good point - I missed checks for first app / last app selection. Let me add that.